src/main/java/net/minecraftforge/client/player/inventory/tabs/GuiTabList.java

```diff
@@ -0,0 +1,193 @@
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+/**
+ * This software is provided under the terms of the Minecraft Forge Public
+ * License v1.0.
+ */
+package net.minecraftforge.client.player.inventory.tabs;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.FontRenderer;
+import net.minecraft.client.gui.GuiButton;
+import net.minecraft.client.gui.inventory.GuiContainer;
+import net.minecraft.client.gui.inventory.GuiInventory;
+import net.minecraft.client.renderer.RenderItem;
+import net.minecraft.client.resources.I18n;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.fml.client.config.GuiUtils;
+
+/**
+ * Enclosing class to keep GuiContainer clean. This handles all the survival tab drawing, and the addition of two buttons.
+ */
+public class GuiTabList
+{
+	private int page;
+	private int maxPages;
+	private GuiContainer container;
+	private List<GuiTab> tabs;
+	private static final ResourceLocation TEXTURE_TABS = new ResourceLocation("textures/gui/container/creative_inventory/tabs.png");
+
+	public GuiTabList(GuiContainer container)
+	{
+		this.container = container;
+		tabs = new ArrayList<GuiTab>(GuiTab.getTabsForGui(container.getClass())); //get a copy of the stored list so it can be modified without consequence
+		maxPages = ((getTabs().size() - 12) / 12) + 1;
+
+		for (GuiTab tab : getTabs())//set the first page
+			if (container.getClass().equals(tab.getTargetGui()))
+			{
+				page = tab.getTabPage();
+				break;
+			}
+	}
+
+	/**
+	 * draws all tabs that are not selected behind the texture
+	 */
+	public void drawTabs(boolean isFront, int mouseX, int mouseY, float partialTicks, int guiLeft, int guiTop, int xSize, int ySize, RenderItem itemRender,
+			FontRenderer fontRendererObj)
+	{
+
+		if (getTabs().size() == 1 && container.getClass().equals(GuiInventory.class)) // don't draw the base tab if it's the only one
+			return;
+
+		int start = page * 12;
+		int end = Math.min(getTabs().size(), (page + 1) * 12);
+
+		for (GuiTab tab : getTabs())
+		{
+			if (tab.getTabIndex() >= start && tab.getTabIndex() < end)
+			{
+
+				if (!tab.isActiveTab(container.getClass()) && !isFront)
+				{
+					Minecraft.getMinecraft().getTextureManager().bindTexture(TEXTURE_TABS);
+					tab.draw(container, isFront, guiLeft, guiTop, xSize, ySize, itemRender, fontRendererObj);
+				}
+
+				if (tab.isActiveTab(container.getClass()) && isFront)
```

Comment:
If you don't want to use a bit operation for those cases, at least put an "else".